packages/datastore/src/sync/processors/mutation.ts

```diff
@@ -307,12 +305,19 @@ class MutationProcessor {
 					this.amplifyConfig
 				);
 
+				if (!customUserAgent) {
+					customUserAgent = { category: Category.DataStore };
+				} else {
+					customUserAgent.category =
+						customUserAgent.category ?? Category.DataStore;
+				}
```

Comment:
Do we mean to update the reference from the parameter directly? Generally speaking, this is not desirable so I want to check if it is intentional

If not intentional I'd suggest just spreading the param so we're not mutating the param, e.g.
```suggestion
				const userAgentDetails = { category: Category.DataStore, ...customUserAgent };
```

Reply:
Thanks Chris, this is the super neat solution I was hoping for!